scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua

```diff
@@ -0,0 +1,47 @@
+-----------------------------------
+-- Area: Empyreal_Paradox
+-- Name: AN
+-- instance 1 Promathia !pos -520 -119 524
+-- instance 2 Promathia !pos 521 -0.500 517
+-- instance 3 Promathia !pos -519 120 -520
+-----------------------------------
+local ID = require("scripts/zones/Empyreal_Paradox/IDs")
+require("scripts/globals/battlefield")
+require("scripts/globals/keyitems")
+require("scripts/globals/missions")
+require("scripts/globals/titles")
+require("scripts/globals/quests")
+-----------------------------------
+
+function onBattlefieldTick(battlefield, tick)
+    tpz.battlefield.onBattlefieldTick(battlefield, tick)
+end
+
+function onBattlefieldRegister(player, battlefield)
+
+end
+
+function onBattlefieldEnter(player, battlefield)
+end
+
+function onBattlefieldDestroy(battlefield)
+end
+
+function onBattlefieldLeave(player, battlefield, leavecode)
+    if leavecode == tpz.battlefield.leaveCode.WON then
+        local name, clearTime, partySize = battlefield:getRecord()
+        player:startEvent(7)
```

Comment:
event 32001 is the victory event on pretty much every battlefield I've seen.  Check other bcnm scripts for use.